subsys/net/ip/tcp.c

```diff
@@ -3065,10 +3065,19 @@ int net_tcp_connect(struct net_context *context,
 		 * net_context_create_ipv4_new() is not called in the packet
 		 * output chain
 		 */
-		ip4 = net_if_ipv4_select_src_addr(
+		if (net_ipv4_is_addr_unspecified(
+			&net_sin(local_addr)->sin_addr))
+		{
+			ip4 = net_if_ipv4_select_src_addr(
 			net_context_get_iface(context),
 			&net_sin(remote_addr)->sin_addr);
```

Comment:
Those lines should get one extra line of indentation since you've indented `net_if_ipv4_select_src_addr()`, currently the function arguments blend with the code.